### DISCUSSION TITLE
Parsing imports into generated AST

### DISCUSSION THREAD

**Discussion by chrismooreajbell** [typescript-eslint/typescript-eslint#8254] [Technical Discussions]
Hi, this is more of a question than a discussion though it may turn in to that.

Say I have a file with a imports that reference other files:
`types.ts`:
```
export type SomeNumberType = number
```
`enums.ts`
```
export enum SomeEnum {
  SomeEnumValue = 'SomeEnumValue'
}
```
`index.ts`
```
export default class SomeClassComponent extends Vue {
  @Prop({ type: String })
  private someEnumProp!: SomeEnum

  @Prop({ type: Number })
  private someNumberProp!: SomeNumberType
  
  ...
}
```

Is there a neat way to grab the original type/enum definitions from `types.ts` and `enums.ts` into the AST for `index.ts`? The problem is in this case both appear as type annotations with no context of the originally exported code.

A use case for this is that I want to add a fixer to do different things depending on if the exported node is a type or an enum. I guess this is just a general TS parsing question. If this is the wrong place to ask I apologies, I could just use a hand and I'm a bit new to all this :sweat_smile:

### Additional Info

Any additional info...

### Before you submit your discussion, please confirm the following. If any of these required steps are not taken, we may not be able to review your RFC. Help us to help you!

- [X] I have [read the discussions guidelines](https://typescript-eslint.io/contributing/discussions) and per those guidelines, this fits a discussion category, _not_ a help request or standard issue.
- [X] I have [searched for related discussions](https://github.com/typescript-eslint/typescript-eslint/discussions) and [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues) and found none that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

**Comment by chrismooreajbell** [comment]
Apologies I've updated the original with a more representative code sample.

**Comment by bradzacher** [comment]
https://typescript-eslint.io/contributing/discussions/

As per our contributing guide for discussions. We do not use them for questions or support requests. The guide includes alternatives.

https://typescript-eslint.io/contributing/issues#questions-and-support-requests
